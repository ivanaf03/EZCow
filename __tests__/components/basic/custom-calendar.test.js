import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import CustomCalendar from "../../../components/basic/custom-calendar";

describe("CustomCalendar", () => {
  const mockDate = new Date("2000-01-01");

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const tree = render(
      <CustomCalendar text="Test" date={mockDate} onChange={jest.fn()} />
    );
    expect(tree).toMatchSnapshot();
  });

  it("should show the calendar when clicking on the button", () => {
    const onChange = jest.fn();
    const tree = render(
      <CustomCalendar text="Test" date={mockDate} onChange={onChange} />
    );
    const button = tree.getByTestId("date-picker-button");
    fireEvent.press(button);
    expect(tree.getByTestId("calendar")).not.toBeNull();
  });
});
