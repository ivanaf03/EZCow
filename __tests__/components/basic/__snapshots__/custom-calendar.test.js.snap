// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCalendar should render correctly 1`] = `
<View
  style={
    [
      {
        "marginBottom": 8,
        "marginLeft": 8,
        "marginRight": 8,
        "marginTop": 8,
      },
      {
        "marginTop": -4,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 14,
          "lineHeight": 20,
        },
        {
          "fontFamily": "Nunito-Light",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
  >
    Test
    :
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
        {
          "backgroundColor": "#ffffff",
        },
        {
          "borderBottomWidth": 2,
          "borderLeftWidth": 2,
          "borderRightWidth": 2,
          "borderTopWidth": 2,
        },
        {
          "borderBottomColor": "#2254ba",
          "borderLeftColor": "#2254ba",
          "borderRightColor": "#2254ba",
          "borderTopColor": "#2254ba",
        },
        {
          "borderBottomLeftRadius": 16,
          "borderBottomRightRadius": 16,
          "borderTopLeftRadius": 16,
          "borderTopRightRadius": 16,
        },
        {
          "fontFamily": "Nunito-Medium",
        },
        [
          {
            "marginTop": 4,
          },
        ],
      ]
    }
    testID="date-picker-button"
  >
    <Text>
      1/1/2000
    </Text>
  </View>
</View>
`;
